src/transport/net_ib.cc

```diff
@@ -500,6 +532,209 @@ static_assert((offsetof(struct ncclIbRecvComm, remFifo) % 32) == 0, "ncclIbSendC
 
 NCCL_PARAM(IbQpsPerConn, "IB_QPS_PER_CONNECTION", 1);
 
+std::unordered_set<std::string> disabledIbPeer;
+void disableIb(std::string peerAddr)
+{
+  if (ncclParamResilientEnabled() && (disabledIbPeer.find(peerAddr) == disabledIbPeer.end()))
+  {
+    disabledIbPeer.insert(peerAddr);
+    usingSocket = true;
+    WARN("NET/IB : Disable the ib peer:%s will use socket for data transmission temperately", peerAddr.c_str());
+  }
+}
+
+bool getIbDisableStatus(std::string peerAddr)
+{
+  return disabledIbPeer.find(peerAddr) != disabledIbPeer.end();
+}
+
+ncclResult_t ncclNetSocketTest(void* request) {
+  struct ncclIbRequest *r = (struct ncclIbRequest*)request;
+  int offset = 0;
+
+  switch (r->type)
+  {
+    case NCCL_NET_IB_REQ_SEND:
+    {
+      // reset the events to correct it for socket transport.
+      r->events = 1;
+      void* host_data = NULL;
+      
+      while (offset < sizeof(int)) NCCLCHECK(ncclSocketProgress(NCCL_SOCKET_SEND, r->sock, &r->send.fifoSlot, sizeof(int), &offset));
+      
+      offset = 0;
+      while (offset < sizeof(int)) NCCLCHECK(ncclSocketProgress(NCCL_SOCKET_SEND, r->sock, &r->send.r, sizeof(int), &offset));
+      
+      offset = 0;
+      while (offset < sizeof(int)) NCCLCHECK(ncclSocketProgress(NCCL_SOCKET_SEND, r->sock, &r->send.size, sizeof(int), &offset));
+      
+      offset = 0;
+      if(r->send.size > 0)
+      {
+        struct cudaPointerAttributes attributes;
+        CUDACHECK(cudaPointerGetAttributes(&attributes, r->send.data));
```

Comment:
>CUDACHECK [](http://example.com/codeflow?start=8&length=9)

does CUDACHECK create a CUDA error instead of NCCL error? Do we need to translate the returned value to NCCL error code?